File: 2022-2023/sourcecode/lab/lab7/NotyJava/app/src/main/java/pl/udu/uwr/pum/notyjava/widget/NotyWidgetService.java
package pl.udu.uwr.pum.notyjava.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

public class NotyWidgetService extends RemoteViewsService {

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new NotyWidgetItemFactory(getApplicationContext(), intent);
    }
}

File: 2022-2023/sourcecode/lab/lab7/NotyKotlin/app/src/main/java/pl/udu/uwr/pum/notykotlin/widget/NotyWidgetService.kt
package pl.udu.uwr.pum.notykotlin.widget

import android.content.Intent
import android.widget.RemoteViewsService

class NotyWidgetService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
        return NotyWidgetItemFactory(applicationContext, intent)
    }
}